### PR TITLE
feat(query): "Request Body Object With Incorrect Media Type" for OpenAPI (#3070)

### LINKED ISSUE #3070: Add "Request Body Object With Incorrect Media Type" query for OpenAPI
### Platform
OpenAPI

### Description
The field 'content' of the request body object should be set to 'multipart' or 'application/x-www-form-urlencoded' when field 'encoding' is set

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares rafaela.soares@checkmarx.com

Closes #3070

**Proposed Changes**
- Added "Request Body Object With Incorrect Media Type" query for OpenAPI. It checks if  the field 'content' of the request body object is not set to 'multipart' or 'application/x-www-form-urlencoded' when field 'encoding' is set

**Considerations**:
- The Request Body Object exists in Components Object and Operation Object.

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/request_body_object_with_incorrect_media_type/query.rego [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		"searchKey": sprintf("components.requestBodies.{{%s}}.content.{{%s}}.encoding.{{%s}}", [r, x]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("components.requestBodies.{{%s}}.content.{{%s}} is 'multipart' or 'application/x-www-form-urlencoded' when 'encoding' is set", [r, x]),
+		"keyActualValue": sprintf("components.requestBodies.{{%s}}.content.{{%s}} is not 'multipart' or 'application/x-www-form-urlencoded' when 'encoding' is set", [r, x]),
+	}
+}
+
+incorrect_media_type(content, media_type) {
+	object.get(content, "encoding", "undefined") != "undefined"
+	regex.match("((^multipart/[a-zA-Z-]+)|application/x-www-form-urlencoded)", media_type) == false
```
Comment (joaoReigota1) [inline assets/queries/openAPI/request_body_object_with_incorrect_media_type/query.rego:39]: can we remove the regex and maybe use the startwith function for performance issues?

**Review by joaoReigota1 — COMMENTED** [review]
Added one comment

File: assets/queries/openAPI/request_body_object_with_incorrect_media_type/query.rego (hunk 1 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/request_body_object_with_incorrect_media_type/query.rego:39]: Yes, of course. Thank you 😄

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**